api/envoy/extensions/filters/listener/http_inspector/v3/http_inspector.proto

```diff
@@ -18,4 +20,11 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message HttpInspector {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.listener.http_inspector.v2.HttpInspector";
+
+  // The number of bytes of the request the filter will initially buffer and
+  // inspect to determine the http version, the filter will iteratively double
+  // the buffer up to 64KiB. If not specified, the initial buffer is 8KiB.
```

Comment:
I'm trying to decouple the interface from what is done by Envoy's filter implementation. I suggest the following:

```suggestion
  // inspect to determine the http version. If not specified, the initial buffer is 8KiB.
  //
  // To reduce the amount of memory that is allocated by default for a request and if the value is larger than 8192 (the default value),
  // the filter will initially allocate an 8KiB buffer, and double that allocation up to the configured value (up to 64KiB).
```

I'm assuming that this is the intent, so feel free to rephrase and clarify.

Reply:
This default window is quite large, and I think it would be good to allow reducing it if a customer is trying to run lean.

Reply:
> This default window is quite large, and I think it would be good to allow reducing it if a customer is trying to run lean.

it could increase dynamically, then may be we can set a default initial size to avoid the complexity of new field. If you think 8kb is two big, 2k or 4k are also fine to me.